public/components/home/home.controller.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+angular.module('textAnalysis')
+    .controller('homeCtrl', ['$scope', '$http', '$anchorScroll', '$location', 'twitterService',
+        function($scope, $http, $anchorScroll, $location, twitterService) {
+
+            $scope.loadingTweets = false;
+
+            $scope.scrollTo = function(id) {
+                $location.hash(id);
+                $anchorScroll();
+            };
+
+            $scope.userInput = {};
+
+            $scope.loadTweets = function() {
+                $scope.loadingTweets = true;
+                twitterService.loadTweets($scope.userInput.text)
+                    .then(function(data) {
+                        $scope.loadingTweets = false;
+                        $location.path('/tweets');
```

Comment:
You're going to have an issue here when you route to the new path, you'll lose access to the tweets you just pulled down.  There are two options, the first is to cache the data in the `twitterService` and when the `/tweets` controller is loaded it calls `twitterService.loadTweets` again which would load from the cache the second time.  The other option is to make it so when the user submits the form it loads a path like: `$location.path('/tweets/' + $scope.userInput.text);`.  You'd have to update your `$routeProvider` config to look for `.when('/tweets/:handle', {...})`.  Then, in the tweet controller you call `twitterService.loadTweets` and use the $routeParams to get the username that was passed.  

In my opinion, the second option is the better of the two.